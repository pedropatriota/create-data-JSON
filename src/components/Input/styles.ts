import styled from "styled-components";

export const Container = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: flex-start;

  label {
    color: #000;
  }
`;

export const InputContainer = styled.div`
  width: 100%;
  height: 46px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background: #f8fbff;
  border: 1px solid #d8dde7;
  border-radius: 4px;
  outline: none;
  padding: 2px 10px;
  span {
    font-weight: 700;
    font-size: 12px;
    color: #858d9f;
    pointer-events: none;
  }
`;
export const Input = styled.input`
  width: 100%;
  height: 46px;
  color: #37383c;
  outline: none;
  border: none;
  background: transparent;
`;
