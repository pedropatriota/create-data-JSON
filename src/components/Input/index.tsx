import React from "react";
import { TValue } from "../../pages/Home/contracts";
import * as Styled from "./styles";

export interface InputProps
  extends React.InputHTMLAttributes<HTMLInputElement> {
  inputValue: (TValue & string) | number | readonly string[] | undefined;
}

const Input = ({ inputValue, ...props }: InputProps) => {
  return (
    <Styled.Container>
      <Styled.InputContainer>
        <Styled.Input value={inputValue} {...props} />
      </Styled.InputContainer>
    </Styled.Container>
  );
};

export default Input;
