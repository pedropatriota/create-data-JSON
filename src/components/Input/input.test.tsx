import { render } from "@testing-library/react";
import user from "@testing-library/user-event";
import Input from ".";

describe("Dropdown component", () => {
  it("should be able to render it properly", async () => {
    const { getByRole } = render(<Input inputValue={45} />);

    const input = getByRole("textbox") as HTMLInputElement;

    expect(input.value).toBe("45");
  });

  it("should be able type it", async () => {
    const mockedOnChange = jest.fn();
    const { getByRole, debug } = render(
      <Input inputValue={45} onChange={mockedOnChange} />
    );

    const input = getByRole("textbox") as HTMLInputElement;
    await user.type(input, "53");

    expect(mockedOnChange).toBeCalled();
  });

  it("should match the snapshot", () => {
    const { container } = render(<Input inputValue={45} />);
    expect(container).toMatchSnapshot();
  });
});
