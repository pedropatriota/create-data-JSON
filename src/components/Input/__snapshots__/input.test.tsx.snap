// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component should match the snapshot 1`] = `
<div>
  <div
    class="sc-aXZVg hARIdx"
  >
    <div
      class="sc-gEvEer jAlFUl"
    >
      <input
        class="sc-eqUAAy ibKdRY"
        value="45"
      />
    </div>
  </div>
</div>
`;
